packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -198,20 +210,36 @@ export class Publisher {
           return batch.resolveEvents()
         }
 
+        let authString = undefined
+        if (this._tokenManager) {
+          const token = await this._tokenManager.getAccessToken()
+          if (token && token.access_token) {
+            authString = `Bearer ${token.access_token}`
+          }
+        }
+
+        const headers: Record<string, string> = {
```

Comment:
This is a little nicer IMO:
```
      const headers: Record<string, string> = {
          'Content-Type': 'application/json',
          'User-Agent': 'analytics-node-next/latest',
          ...(authString ? { Authorization: authString } : {})
        }
```

Reply:
This looks like the existing code! (I pasted into my IDE and it agrees :smile:)